### ISSUE TITLE
关于rollout_torchengine计算full_mask

### ISSUE THREAD

**Issue by kustcl** [jingyaogong/minimind#767]
大佬你好 ，在源码所实现的generate函数中，对于批量生成的样本，为了确保长度一致，是使用eos_token_id进行填充的的 代码如下：
if eos_token_id is not None: next_token = torch.where(finished.unsqueeze(-1), next_token.new_full((next_token.shape[0], 1),eos_token_id), next_token)
而在rollout_torch_engine 中  对于采样完成后生成计算新的attention_mask是 ：full_mask = (output_ids != self.tokenizer.pad_token_id).long()  
这样是不是会 让eos_token_id也参与到logprob的计算中

**Comment by jingyaogong** [comment]
❌
算出来不等于参与 loss
第一个 eos 该算，后面补齐出来的 eos 会被 completion_mask 截掉
